cpp/dolfinx/io/XDMFFile.h

```diff
@@ -104,9 +103,9 @@ class XDMFFile
   /// @param[in] xpath XPath where Mesh Grid is located
   /// @return A Mesh distributed on the same communicator as the
   ///   XDMFFile
-  mesh::Mesh read_mesh(const fem::CoordinateElement& element,
-                       mesh::GhostMode mode, std::string name,
-                       std::string xpath = "/Xdmf/Domain") const;
+  mesh::Mesh<double> read_mesh(const fem::CoordinateElement& element,
```

Comment:
I guess your comment:

> No changes in the Python interface. Will later add support for meshes with different scalar types from Python.

applies here as well?

Reply:
We'll deprecate XDMF at some point, hence I wasn't planning to update it for different float types.